src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -635,3 +637,25 @@ def _configure_security_policy_storage_params(cmd):
                options_list=('--end-ip-address',),
                help='The end IP address of the firewall rule. Must be IPv4 format. Use value'
                ' \'0.0.0.0\' to represent all Azure-internal IP addresses.')
+
+
+#####
+#           sql server vnet-rule
+#####
+
+with ParametersContext(command='sql server vnet-rule') as c:
+    # Help text needs to be specified because 'sql server vnet-rule create' is a custom
+    # command.
+    c.argument('server_name',
+               options_list=('--server', '-s'), completer=get_resource_name_completion_list('Microsoft.SQL/servers'))
+
+    c.argument('virtual_network_rule_name',
+               options_list=('--name', '-n'))
+
+    c.argument('virtual_network_subnet_id',
+               options_list=('--subnet'),
+               help='Name or ID of the subnet that allows access to an Azure Sql Server. '
+               'If subnet name is provided, --vnet-name must be provided.')
+
+register_extra_cli_argument('sql server vnet-rule create', 'vnet_name', options_list=('--vnet-name'),
+                            help='The virtual network name', validator=validate_subnet)
```

Comment:
Look at you getting all tricky! Ok cool stuff.

Reply:
Please test both modes

Reply:
And add examples of both modes to _help.py

Reply:
Tested both and added examples to _help.py